core/frontend/src/IModelConnection.ts

```diff
@@ -1078,21 +1078,61 @@ export namespace IModelConnection {
 
   /** The collection of [[CodeSpec]] entities for an [[IModelConnection]]. */
   export class CodeSpecs {
-    private _loaded?: CodeSpec[];
-
     /** @internal */
     constructor(private _iModel: IModelConnection) { }
 
-    /** Loads all CodeSpec from the remote IModelDb. */
-    private async _loadAllCodeSpecs(): Promise<void> {
-      if (this._loaded)
-        return;
+    private _isCodeSpecProperties(x: any): x is CodeSpecProperties {
+      if (!x || !x.scopeSpec || !Object.values(CodeScopeSpec.Type).includes(x.scopeSpec.type)) return false;
 
-      this._loaded = [];
-      const codeSpecArray: any[] = await IModelReadRpcInterface.getClientForRouting(this._iModel.routingContext.token).getAllCodeSpecs(this._iModel.getRpcProps());
-      for (const codeSpec of codeSpecArray) {
-        this._loaded.push(CodeSpec.createFromJson(this._iModel, Id64.fromString(codeSpec.id), codeSpec.name, codeSpec.jsonProperties));
-      }
+      if (typeof x.scopeSpec.fGuidRequired !== "boolean" && typeof x.scopeSpec.fGuidRequired !== "undefined")
+        return false;
+
+      if (typeof x.scopeSpec.relationship !== "string" && typeof x.scopeSpec.relationship !== "undefined") return false;
+
+      if (typeof x.spec?.isManagedWithDgnDb !== "boolean" && typeof x.spec?.isManagedWithDgnDb !== "undefined")
+        return false;
+
+      if (typeof x.version !== "string" && typeof x.version !== "undefined") return false;
+
+      return true;
+    }
+
+    /** Loads the needed CodeSpec from the remote IModelDb. */
+    private async _loadCodeSpec(identifier: { name: string; id: never } | { id: string; name: never }): Promise<CodeSpec> {
+      const query = `
+        SELECT ECInstanceId AS Id, Name, JsonProperties
+        FROM BisCore.CodeSpec
+        WHERE ${identifier.name !== undefined ? ` Name='${identifier.name}'` : ` Id=${identifier.id}`}`;
```

Comment:
I might be wrong here, but passing the name and id separately from the query should make it cache better on the backend. Not very useful for the cloud due to horizontal scaling, but might be helpful for electron/mobile.